### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by maths29**
In this project, I use the Decorator design pattern to correct the names of people.

- Implement composition (as an example of the design pattern).
- Run a program using the command line.

### CODE REVIEW

File: capitalize.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+required_relative 'decorator'
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.upcase
```
Comment (Ibizugbe) [inline capitalize.rb]: Nice job so far👍🏼. From the requirement, the correct name should be capitalized and not converted to uppercase as you have implemented😉. Kindly fix this👍🏼.

```
 @nameable.correct_name.capitalize
```


image

File: trimcase.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require_relative 'decorator'
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.slice(0..10)
```
Comment (Ibizugbe) [inline trimcase.rb]: Nice job implementing the correct name trimmer😊. but it is required that a maximum of 10 characters should be retained after being trimmed. but yours currently will allow a maximum of 11 characters 😉. Kindly see the suggestion below.

```
  @nameable.correct_name.slice(0..9)
```

File: Person_rb [hunk 3]
```diff
@@ -1,3 +1,4 @@
+required_relative 'nameable'
```
Comment (Ibizugbe) [inline Person_rb]: Just as required, kindly ensure the person class inherits from the nameable class😉.
image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @maths29 👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] a `Namable` class with a method 'correct_name' that raises an error has been implemented👍🏼.
- [x] the constructor a namable object from params is assigned to an instance variable👍🏼.


## Required Changes ♻️

- [ ] Nice job so far👍🏼. a couple of your files are not having the right extensions `.rb` (person, student and teacher ) kindly do well to name them appropriately and ensure your decorators are working well👍🏼.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @maths29 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Capitalize & Trimmer decorator classes work as expected ✔
- Person class inherits from Nameable ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a good summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**